sys/syslog/syslog.c

```diff
@@ -0,0 +1,80 @@
+#include <stdarg.h>
+#include <string.h>
+#include <unistd.h>
+#include <time.h>
+#include <netdb.h>
+
+#define ENABLE_DEBUG (0)
+#include "debug.h"
+
+#include "syslog/syslog.h"
+#include "xtimer.h"
+
+static char buf[SYSLOG_BUF_NUMOF][SYSLOG_MAX_LEN];
+static syslog_msg_t _msg[SYSLOG_BUF_NUMOF];
+static char _ident[IDENT_MAX_LEN];
+static char hostname[HOST_NAME_MAX];
+static mutex_t syslog_mutex = MUTEX_INIT;
+static int _facility = 0;
```

Comment:
If you like, you can drop the explicit initialization of  `_facility`, since it will be initialized to `0` anyway. See sections 6.2.4 (3) and 6.7.8 (10) of a [draft of the ISO C (ANSI C) standard](http://www.open-std.org/JTC1/SC22/WG14/www/docs/n1256.pdf).

Reply:
Yea that is true. But it should be auto-optimized by the compiler anyway? Or is it just always good practice to drop such redundant initializations? I asking bc I find it a little bit more readable with an explicit initialization.

Reply:
True, the compiler does not care whether one writes the additional assignment.

I would drop it, since for me it makes the code easier to read but I am not sure if it is considered best practice and I am not aware of any related RIOT-OS guideline. One might argue that omitting the explicit initialization hides what is happening, making the code more difficult to read.

I guess, I just wanted to point it out and see what you think of it, since you explicitly initialized `_facility` but not e.g. `_ident`. But to be clear, I don't think it really matters. This was just my curiosity speaking.

Reply:
Ok, I was curious if there are any implications I am not aware of. Anyways, it should be at least consistent in my code. I will have a look into it (Maybe zero isn't even the best idea). Thanks for your feedback!